### PR TITLE
replace Cypress with Playwright

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Bit of a big one. This removes more or less all traces of cypress, and switches over to using playwright instead. We didn't replace all existing test with playwright equivalents, because the majority of all existing tests were not actually run in CI.

We've been looking to move over to Playwright for quite a while, but with the recent e2e test and cypress install issues we got a bit of a kick in the butt. Ideally we would've wanted to immediately move over to a setup where we run the backend for E2E tests too, but that will have to wait for some breathing room.

The new setup runs tests from the root rather than in the app. Any package with an `e2e-tests` folder with be included. We do consider this setup to be somewhat experimental for now, although there's no real API surface to maintain other than the utils and what create-app looks like.

To help out with Git archaeology we've removed all existing tests in a single commit: https://github.com/backstage/backstage/commit/a07764f44accb4be271ad005a08014dd6ab499f0.
CC @backstage/discoverability-maintainers @backstage/techdocs-maintainers, let us know if you have any questions on how to restore these tests if you want to do that. For now we had to remove them all to completely get rid of the cypress install.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/create-app/src/createApp.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export default async (opts: OptionValues): Promise<void> => {
   const templateDir = opts.templatePath
     ? paths.resolveTarget(opts.templatePath)
     : paths.resolveOwn('templates/default-app');
-  const tempDir = resolvePath(os.tmpdir(), answers.name);
+  const tempDir = resolvePath(os.tmpdir(), answers.name); // TODO: ...
```
Comment (benjdlambert) [inline packages/create-app/src/createApp.ts:71]: hmm?

**Review by benjdlambert — APPROVED** [review]
Looks good to me, just the one comment about the pending todo:?

File: .changeset/twelve-clouds-design.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/create-app': patch
+---
+
+The E2E test setup based on Cypress has been replaced with one based on [Playwright](https://playwright.dev/). Migrating existing apps is not required as this is a standalone setup, only do so if you also want to switch from Cypress to Playwright.
+
+The scripts to run the E2E tests have been removed from `packages/app/package.json`, as they are now instead run from the root. Instead, a new script gas been added to the root `package.json`, `yarn test:e2e`, which runs the E2E tests in development mode, unless `CI` is set in the environment.
```
Comment (freben) [inline .changeset/twelve-clouds-design.md:7]:
```suggestion
The scripts to run the E2E tests have been removed from `packages/app/package.json`, as they are now instead run from the root. Instead, a new script has been added to the root `package.json`, `yarn test:e2e`, which runs the E2E tests in development mode, unless `CI` is set in the environment.
```

File: packages/e2e-test-utils/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+_This package was created through the Backstage CLI_.
+
+## Installation
+
+Install the package via Yarn:
+
+```sh
+cd <package-dir> # if within a monorepo
+yarn add @backstage/e2e-test-utils
```
Comment (freben) [inline packages/e2e-test-utils/README.md:11]:
```suggestion
yarn add --dev @backstage/e2e-test-utils
```

File: packages/e2e-test-utils/package.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    "build": "backstage-cli package build",
+    "lint": "backstage-cli package lint",
+    "test": "backstage-cli package test",
+    "clean": "backstage-cli package clean",
+    "prepack": "backstage-cli package prepack",
+    "postpack": "backstage-cli package postpack"
+  },
+  "dependencies": {
+    "@manypkg/get-packages": "^1.1.3",
+    "@playwright/test": "^1.32.3",
```
Comment (freben) [inline packages/e2e-test-utils/package.json:39]: should we make this a peer?

File: packages/e2e-test-utils/src/playwright.ts [hunk 5]
```diff
@@ -13,5 +13,5 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/// <reference types="cypress" />
-import './commands';
+
+export * from './playwright/index';
```
Comment (freben) [inline packages/e2e-test-utils/src/playwright.ts:17]: Do we need this file? Don't we use the dedicated export point anyway

Comment (Rugvip) [reply packages/e2e-test-utils/src/playwright.ts:17]: Yeah turns our entry point detection in repo-tools didn't like the folder entry, will fix, but later :p

**Comment by jamieklassen** [comment]
@Rugvip is there some more thorough documentation for this decision (I'm optimistically imagining some links to issues/discussions/discord/meeting minutes)? My company runs some cypress e2e tests for our custom plugin functionality; wondering if we might encounter similar "e2e test and cypress install issues", or whether Playwright has some distinct advantages that could save us CI resources or maintenance burden? This might incentivize us to make a similar change in testing strategy.

**Comment by Rugvip** [comment]
@jamieklassen nothing public unfortunately, we tried out playwright in an internal Backstage projects after weighing alternatives, and liked the result. The original decision to use cypress in Backstage OSS wasn't particularly carefully made, it was basically what we were using internally at the time of OSS, which in turn had already been in use for a while.

As mentioned here the e2e framework isn't particularly closely tied to Backstage at all, you can really run whatever you want, we just like to provide some setup out of the box. The `e2e-tests` folder convention implemented here is a quite loose convention that we chose as some form of sane default, but imagine that people can chose their own there too.

If you want to be have some reasons to use Playwright, try out the `--ui` mode in the playwright CLI for writing and debugging tests, especially along with a dev server running on the side serving the frontend as we do for the local setup.

**Pull request merged**